protocols/virtualfund/virtualfund_single_hop_test.go

```diff
@@ -172,26 +172,29 @@ func TestClone(t *testing.T) {
 	}
 }
 
-func collectPeerSignaturesOnSetupState(V *channel.SingleHopVirtualChannel, myRole uint, prefund bool) {
+func collectPeerSignaturesOnSetupState(v *channel.SingleHopVirtualChannel, myRole uint, prefund bool) *channel.SingleHopVirtualChannel {
 	var state state.State
 	if prefund {
-		state = V.PreFundState()
+		state = v.PreFundState()
 	} else {
-		state = V.PostFundState()
+		state = v.PostFundState()
 	}
```

Comment:
This code is correct, but it made me think / investigate a bit to convince myself that no side effects leaked onto `v`.

I'm not allowed to make this suggestion, but something like 

```
func collectPeerSignaturesOnSetupState(v channel.SingleHopVirtualChannel, myRole uint, prefund bool) *channel.SingleHopVirtualChannel {
	withSigs := v.Clone()

	var state state.State
	if prefund {
		state = withSigs.PreFundState()
	} else {
		state = withSigs.PostFundState()
	}

	if myRole != alice.Role {
		aliceSig, _ := state.Sign(alice.PrivateKey)
		withSigs.AddStateWithSignature(state, aliceSig)
	}
	if myRole != p1.Role {
		p1Sig, _ := state.Sign(p1.PrivateKey)
		withSigs.AddStateWithSignature(state, p1Sig)
	}
	if myRole != bob.Role {
		bobSig, _ := state.Sign(bob.PrivateKey)
		withSigs.AddStateWithSignature(state, bobSig)
	}
	return withSigs
}
```

is slightly more explicit.